plugins/utils/navbuddy.nix

```diff
@@ -0,0 +1,273 @@
+            with types;
+              either
+              percentageType
+              (submodule {
+                options = {
+                  height = mkPercentageOpt 40 "The height size (in %).";
+
+                  width = mkPercentageOpt 100 "The width size (in %).";
+                };
+              })
+          )
+          "60"
+          "The size of the window.";
+
+        position =
+          helpers.defaultNullOpts.mkNullable
+          (
+            with types;
+              either
+              percentageType
+              (submodule {
+                options = {
+                  height = mkPercentageOpt 40 "The height size (in %).";
+
+                  width = mkPercentageOpt 100 "The width size (in %).";
+                };
+              })
+          )
+          "50"
+          "The position of the window.";
+
+        scrolloff = helpers.mkNullOrOption types.int ''
+          scrolloff value within navbuddy window
+        '';
+
+        sections = {
+          left = {
+            size = mkPercentageOpt 20 "The height size (in %).";
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+          };
+
+          mid = {
+            size = mkPercentageOpt 40 "The height size (in %).";
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+          };
+
+          right = {
+            border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" null ''
+              "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+              starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+            '';
+            preview = helpers.defaultNullOpts.mkEnum ["leaf" "always" "never"] "leaf" ''
+              Right section can show previews too.
+                    Options: "leaf", "always" or "never"
+            '';
+          };
+        };
+      };
+
+      nodeMarkers = {
+        enabled = helpers.defaultNullOpts.mkBool true;
+        icons = {
+          leaf = helpers.defaultNullOpts.mkStr "  " ''
+            The icon to use for leaf nodes.
+          '';
+
+          leaf_selected = helpers.defaultNullOpts.mkStr " → " ''
+            The icon to use for selected leaf node.
+          '';
+          branch = helpers.defaultNullOpts.mkStr "  " ''
+            The icon to use for branch nodes.
+          '';
+        };
```

Comment:
```suggestion
        enabled = helpers.defaultNullOpts.mkBool true "Enable node markers.";

        icons = {
          leaf = helpers.defaultNullOpts.mkStr "  " ''
            The icon to use for leaf nodes.
          '';

          leaf_selected = helpers.defaultNullOpts.mkStr " → " ''
            The icon to use for selected leaf node.
          '';

          branch = helpers.defaultNullOpts.mkStr "  " ''
            The icon to use for branch nodes.
          '';
        };
```

Reply:
Also, `leaf_selected` -> `leafSelected`. We use camelCase for all of our options.
Don't forget to write `leaf_selected = leafSelected` in the `config` section below.

Reply:
ahh yes ofc course will add this change then update in setup options

Reply:
And don't forget to change it in the test.